app/controllers/ATSPAYEDataController.scala

```diff
@@ -42,17 +42,25 @@ class ATSPAYEDataController @Inject()(npsService: NpsService, payeAuthAction: Pa
 
   def getATSDataMultipleYears(nino: String, yearFrom: Int, yearTo: Int): Action[AnyContent] = payeAuthAction.async {
     implicit request =>
-      def dataList: Seq[Future[Option[JsValue]]] = (yearFrom to yearTo).toList map { year =>
+      def dataList: Seq[Future[Either[Int, JsValue]]] = (yearFrom to yearTo).toList map { year =>
         callConnector(nino, year) map {
-          case Right(response) => Some(Json.toJson(response))
+          case Right(response) => Right(Json.toJson(response))
           case Left(error) =>
             logger.error(s"Fetching $year data for $nino returned ${error.status}")
-            None
+            Left(error.status)
         }
       }
-      Future.sequence(dataList) map { data =>
-        val flattenedData = data.flatten
-        if (flattenedData.isEmpty) NotFound(s"No data found for $nino") else Ok(Json.toJson(flattenedData))
+
+      Future.sequence(dataList) map { seqEither =>
+        val seqJsValue = seqEither.filter(either => either.isRight) flatMap { right =>
+          right.toOption
+        }
+
+        if (seqJsValue.nonEmpty) Ok(Json.toJson(seqJsValue))
+        else if (seqEither.contains(Left(NOT_FOUND))) NotFound(s"No data found for $nino")
+        else if (seqEither.contains(Left(BAD_REQUEST))) BadRequest(s"Bad request for $nino")
+        else InternalServerError
```

Comment:
This is an informative error, it would be pretty difficult to debug this status code with no extra logging

Reply:
done